pkg/ebpf/tracer.go

```diff
@@ -318,6 +323,15 @@ func (t *Tracer) GetActiveConnections(clientID string) (*Connections, error) {
 
 	conns := t.state.Connections(clientID, latestTime, latestConns)
 	names := t.reverseDNS.Resolve(conns)
+
+	for key, dnsStats := range t.reverseDNS.GetDNSStats() {
+		if !t.config.CollectLocalDNS && key.serverIP.IsLoopback() {
+			continue
+		}
+		t.state.StoreDNSStats(key, dnsStats)
+	}
```

Comment:
Can we move the calls to StoreDNSStats above the calls to Connections and change Connections to do all the DNS adding the AddDNSStats is doing ?

Reply:
I initially had it exactly that way. The problem was that for new clients, there was no local `dnsStats` to use in the code block that we have for `AddDNSStats`. There are a few options I can consider:
1. Ignore this since it happens only on the first call for a new client
2. Instead `StoreDNSStats`, we also pass a `clientID` and then in `StoreDNSStats`, we create one if it does not exist (just like what we do in `Connections`)
3.  Have a global copy of `DNSStats` in `NetworkState` that new clients can fall back to.

Please suggest if I am missing something obvious.

Reply:
Could we have `Connections()` take the most recent  copy of the DNS stats and copy them into each client state (as well as initialize any clientState that doesn't exist)?